build.gradle

```diff
@@ -132,6 +139,26 @@ task jacocoTestQuickReport(type: JacocoReport, dependsOn: testQuick) {
 	executionData(testQuick)
 }
 
+shadowJar {
+	// Be careful when updating jars - you may want to set the duplicates strategy to WARN
+	// to see if any of the jars are shadowing the others when building the fat jar, which
+	// has been the case in the past
+	dependsOn buildDocs
+	duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+	archiveAppendix = 'test-shadow'
+	from sourceSets.test.output
+	// we don't want to package the auth shadow jar, so no test runtime classpath
+	configurations = [project.configurations.runtimeClasspath]
+
+	enableRelocation true
+	relocationPrefix 'us.kbase.workspace.shadow'
+	
+	mergeServiceFiles()
+	
+	from(BUILD_JAVA_DOC_DIR) { into "$IN_JAR_DOCS_DIR/javadoc" }
+	from(BUILD_OTHER_DOC_DIR) { into IN_JAR_DOCS_DIR }
```

Comment:
Are these two lines just copy files into /server_docs?

Reply:
yep

Reply:
so they are in shadow jar. Why we need to do this? Or is just for documentation.

Reply:
So that if someone runs a test against the shadowjar that touches the /docs endpoint for some reason things work as expected

Reply:
how are /docs and /server_docs connected? It somehow knows to look into /server_docs when a test touches /docs endpoint?

Reply:
👍